src/solver/td3.ml

```diff
@@ -496,16 +489,19 @@ module Base =
         (* solve x Widen *)
       in
 
-      let rec destabilize_normal x =
+      let rec destabilize_normal ?(notify_destabilized = fun _ _ _ -> ()) x =
```

Comment:
Since the fun argument does not change between invocations it might make sense to have an inner recursive function that does not have this parameters but uses the one from the outside scope.